lib/src/functional_value_notifiers.dart

```diff
@@ -32,15 +32,6 @@ abstract class FunctionalValueNotifier<TIn, TOut> extends ValueNotifier<TOut> {
     super.addListener(listener);
   }
 
-  @override
```

Comment:
hmmm, this is in the end a question if these FunctionalValueNotifiers should still update even if they don't have any listeners, right?
So the better approach would be to add an option parameter `keepChainAlive == false` that would prevent the removal of the previous listeners. 
In the majority of cases you can't to remove all listeners to avoid memoryLeaks

Reply:
Yes, but now we have unexpected behavior. When I create a listener via map(...), I can get the actual value without listening until I add the listener and remove it. 

If I map ValueNotifier and never listen, can it cause memoryLeaks? Now MapValueNotifier always subscribes to parent in the constructor.

In my opinion, the behavior of the value field should not change. Maybe we need to always listen to parent ValueListener. All subscriptions will be removed when root ValueNotifier is disposed of.

Reply:
That exactly is the problem. When the root ValueNotifier gets disposed the listeners don't get informed about the disposal. I have to test if the GC is able to collect everything in that case.

Reply:
hmm, preliminary tess with the memory analyser shows that if the containing object can get garbage collected it seems that all valuenotifier get released.

So maybe we just invert the default parameter to `disposeChain=false` so that you can force the disposal but you don' have to?

Reply:
ok, I now ran into the same problem as you and probably the only way is to remove the removeListener override